shared/src/metabase/mbql/normalize.cljc

```diff
@@ -251,8 +251,11 @@
                  (assoc :name tag-name))])))
    template-tags))
 
-(defn- normalize-query-parameter [{:keys [type target], :as param}]
+(defn normalize-query-parameter
+  "Normalize a parameter in the query `:parameters` list."
+  [{:keys [type target id], :as param}]
   (cond-> param
+    id     (update :id mbql.u/qualified-name)
```

Comment:
Parameter `:id` wasn't getting normalized to a string like everything else so if you accidentally passed it in as a keyword it would cause a schema validation error